codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -20,83 +20,128 @@
 import io.netty.handler.codec.compression.ZlibWrapper;
 import io.netty.util.internal.StringUtil;
 
+import java.util.regex.Pattern;
+
 /**
  * Compresses an {@link HttpMessage} and an {@link HttpContent} in {@code gzip} or
- * {@code deflate} encoding while respecting the {@code "Accept-Encoding"} header.
- * If there is no matching encoding, no compression is done.  For more
- * information on how this handler modifies the message, please refer to
- * {@link HttpContentEncoder}.
+ * {@code deflate} encoding while respecting the {@code "Accept-Encoding"} header. If there is no matching encoding,
+ * no compression is done.
+ *
+ * Note that only a {@link FullHttpResponse} or a response with {@code "Transfer-Encoding: chunked"}
+ * will be compressed. Also the {@code "Content-Length"} and {@code "Content-Type"} are checked against
+ * the given <tt>minCompressableContentLength</tt> and <tt>compressableContentTypes</tt> to determined if
+ * a compression would be effective at all.
+ *
+ * For more information on how this handler modifies the message, please refer to {@link HttpContentEncoder}.
  */
 public class HttpContentCompressor extends HttpContentEncoder {
 
+    public static final int DEFAULT_COMPRESSION_LEVEL = 6;
+    public static final int DEFAULT_WINDOW_BITS = 15;
+    public static final int DEFAULT_MEM_LEVEL = 8;
+    public static final int DEFAULT_MIN_COMPRESSABLE_CONTENT_LENGTH = 1024;
+    public static final Pattern DEFAULT_COMPRESSABLE_CONTENT_TYPES = Pattern.compile("(text/.*|application/json.*)");
```

Comment:
I understand, for this very reason I left the beaking unit test in the PR. I think backward compatibility is the way to go! Shall I update the PR with the logic I proposed earlier? I'd also add / expand the JavaDoc so that a new user of the class is pushed in the right direction. You still could, at some point, then deprecate the old constructor(s). This would show existing users of the class that there is a change and they could figure out which settings to use...